components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImplTestCase.java

```diff
@@ -558,17 +559,23 @@ public void testUpdateNullAPI() throws APIManagementException {
     @Test(description = "Update api status")
     public void testUpdateAPIStatus() throws APIManagementException, LifecycleException {
         ApiDAO apiDAO = Mockito.mock(ApiDAO.class);
+        WorkflowDAO workflowDAO = Mockito.mock(WorkflowDAO.class);
         APILifecycleManager apiLifecycleManager = Mockito.mock(APILifecycleManager.class);
         APIPublisherImpl apiPublisher = new APIPublisherImpl(user, apiDAO, null, null, null, apiLifecycleManager, null,
-                null);
+                workflowDAO);
         API api = SampleTestObjectCreator.createDefaultAPI().build();
         String uuid = api.getId();
         String lifecycleId = api.getLifecycleInstanceId();
         Mockito.when(apiDAO.getAPI(uuid)).thenReturn(api);
         LifecycleState lifecycleState = SampleTestObjectCreator.getMockLifecycleStateObject(lifecycleId);
+        Mockito.when(apiLifecycleManager.getCurrentLifecycleState(lifecycleId)).thenReturn(lifecycleState);       
         Mockito.when(apiLifecycleManager
                 .executeLifecycleEvent(APIStatus.CREATED.getStatus(), APIStatus.PUBLISHED.getStatus(), lifecycleId,
                         user, api)).thenReturn(lifecycleState);
+        API.APIBuilder apiBuilder = new API.APIBuilder(api);
+        apiBuilder.lifecycleState(lifecycleState);
+        apiBuilder.updatedBy(user);
+        api = apiBuilder.build();        
```

Comment:
Above 4 lines is repeated 4 times in this test case. Better to remove it to a private method